### PR TITLE
feat(ecs-patterns): dualstack ALB

### LINKED ISSUE #29039: ApplicationLoadBalancedFargateService : Add Support for IPv6 on LB's
### Describe the feature

I would like to enable IPv6 support on the load balancers created by ApplicationLoadBalancedFargateService and ApplicationLoadBalancedEc2Service.

Ideally this could be used as a new property for these two classes.``

### Use Case

With the introduction of AWS charging for public IPv4 addresses ([source](https://aws.amazon.com/blogs/aws/new-aws-public-ipv4-address-charge-public-ip-insights)), I am looking to enable IPv6 addresses on my ECS ALBs.

### Proposed Solution

In the `elbv2.ApplicationLoadBalancer` class, there is a property for ipv6: 
```
new elbv2.ApplicationLoadBalancer(stack, 'ALB', {
  vpc,
  internetFacing: true,
  ipAddressType: elbv2.IpAddressType.DUAL_STACK,
});
```

My proposal would be to add the property `ipAddressType` to ApplicationLoadBalancedFargateService and ApplicationLoadBalancedEc2Service with the corresponding actions to enable when necessary. The default would be `elbv2.IpAddressType.IPV4` which would make this an opt-in service.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.126.0

### Environment details (OS name and version, etc.)

MacOS Monterey 12.7

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #29039.

### Reason for this change

Both `ApplicationLoadBalancedFargateService` and `ApplicationLoadBalancedEc2Service` don't support specifying dualstack ALB.

### Description of changes

Added `ipAddressType` to `ApplicationLoadBalancedServiceBaseProps`.

### Description of how you validated changes

Added both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jmnarloch** [comment]
I would be interested in this change, is there anything I could help with?

**Comment by badmintoncryer** [comment]
@jmnarloch Thanks for your comments. This PR waits for community review and I think it will take some weeks or months. Please wait a moment!

**Comment by badmintoncryer** [comment]
@teankie It depends on the community reviewer and maintainer. Please 👍 to the [issue](#29039) to increase the priority.

**Comment by pahud** [comment]
Hi

This PR has been pending for community review for a while. Please consider posting it on the #contributing channel in cdk.dev. Community members will be on the lookout there as well for possible reviews.

Check [How to get your P2 PR community reviewed](https://github.com/aws/aws-cdk/wiki/CDK-Community-PR-Reviews#how-to-get-your-p2-pr-community-reviewed) for more details.

**Review by tmokmss — APPROVED** [review]
This PR looks just to add a property in the same way as #30069. There seem to be two more constructs to be updated (ApplicationMultipleTargetGroupsServiceBase and NetworkMultipleTargetGroupsServiceBase).

**Comment by badmintoncryer** [comment]
@tmokmss Thank you for your review. I'll try to create PR for those constructs later😉

**Comment by badmintoncryer** [comment]
@pahud 
Hi

Thanks to my post on cdk.dev, this PR successfully passed the community review.
I would appreciate it if someone could conduct a maintainer review.

Thank you in advance.

File: packages/aws-cdk-lib/aws-ecs-patterns/README.md [hunk 1]
```diff
@@ -1123,7 +1123,7 @@ const queueProcessingFargateService = new ecsPatterns.NetworkLoadBalancedFargate
 });
 ```
 
-### Use dualstack NLB
+### Use dualstack Load Balancer
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-ecs-patterns/README.md:1126]: Can you have subheading with Network Load Balancer and Application Load Balancer and add some description please?

**Review by GavinZZ — COMMENTED** [review]
LGTM, just a slight doc update would make it perfect

File: packages/aws-cdk-lib/aws-ecs-patterns/README.md (hunk 1 above)
Comment (badmintoncryer) [reply packages/aws-cdk-lib/aws-ecs-patterns/README.md:1126]: I've updated docs. Could you please confirm it again?

**Review by GavinZZ — APPROVED** [review]
LGTM, appreciate it!

**Pull request merged**